recipes/superlu_dist/meta.yaml

```diff
@@ -0,0 +1,56 @@
+{% set version = "5.2.2" %}
+
+{% set variant = "openblas" %}
+
+package:
+  name: superlu_dist
+  version: {{ version }}
+
+source:
+  fn: superlu-{{ version }}.tar.gz
+  url: http://crd-legacy.lbl.gov/~xiaoye/SuperLU/superlu_dist_{{ version }}.tar.gz
+  sha256: bd13e694bae1d703e66d8efd8f67e31fb0b68466b248a37ed165b371945d49ca
+  patches:
+    - shared-static.patch
```

Comment:
Usually, cmake projects are built twice for static and shared libs

Reply:
That's really weird! This ought to be much more efficient, since a single build produces both targets. This seems to be a well-documented pattern on various CMake lists, etc. Is there a reason to do the full build & install twice instead of using this simple fix?

Reply:
> This seems to be a well-documented pattern on various CMake lists, etc.

where?

> Is there a reason to do the full build & install twice instead of using this simple fix?

In this case, it's a simple fix (Your fix will not work on windows as the static lib and the import library of the shared lib will have the same name and clash), but usually it's not the case.

Reply:
Also, the fix above is not really different doing the build twice, as the code will be built twice for the two libraries. (You'll need to use OBJECT libraries to avoid building twice)

Reply:
> where?

- [here](https://cmake.org/pipermail/cmake/2007-March/013053.html)
- [here](https://www.mail-archive.com/cmake@cmake.org/msg07864.html)
- [here](https://stackoverflow.com/questions/2152077/is-it-possible-to-get-cmake-to-build-both-a-static-and-shared-version-of-the-sam)

>  the code will be built twice for the two libraries

Ah, that is inefficient. It's astonishing to me that this isn't CMake's default behavior. Still, it avoids either:

1. running installation twice, or
2. manually staging the static lib instead of doing a normal installation

I'll revert it to the inefficient, undocumented double-build approach, so we don't need to ship a patch.